src/functions.php

```
@@ -399,58 +399,53 @@ function run(
-                $password = askHiddenResponse(" [sudo] password for {{remote_user}}: ");
-            }
-            $run("echo -e '#!/bin/sh\necho \"\$PASSWORD\"' > $askpass");
-            $run("chmod a+x $askpass");
-            $command = preg_replace('/^sudo\b/', 'sudo -A', $command);
-            $output = $run(" SUDO_ASKPASS=$askpass PASSWORD=%sudo_pass% $command", $runParams->with(
-                secrets: ['sudo_pass' => escapeshellarg($password)],
-            ));
-            $run("rm $askpass");
-            return $output;
+    if (Deployer::isWorker()) {
+        $response = Deployer::masterCall($host, 'run', $runParams);
```

PHPStan (2.1.15)

Parameter #2 $host of static method Deployer\Deployer::masterCall() expects Deployer\Host\Host, string given.


(argument.type)